src/matter/cluster/BasicInformationCluster.ts

```diff
@@ -28,29 +28,36 @@ const CapabilityMinimaT = ObjectT({
  * This cluster provides attributes and events for determining basic information about Nodes, which supports both
  * commissioning and operational determination of Node characteristics, such as Vendor ID, Product ID and serial number,
  * which apply to the whole Node. Also allows setting user device information such as location.
- * 
+ *
  * @see {@link MatterCoreSpecificationV1_0} § 11.1
  */
 export const BasicInformationCluster = Cluster({
     id: 0x28,
     name: "Basic Information",
 
+    /* TODO We need to add the global Attributes/Elements, see Core Spec 7.13 - with new structure maybe like that?
+            And yes some attributes could be automatically generated from the registered command implementations or such.
+    globalClusterAttributes: {
+        clusterRevision: 1
+    },
+    */
+
     /** @see {@link MatterCoreSpecificationV1_0} § 11.1.6.3 */
     attributes: {
         /** Revision number of the Data Model against which the Node is certified */
-        dataModelRevision: Attribute(0, UnsignedIntT),
+        dataModelRevision: Attribute(0, BoundedUnsignedIntT({ min: 0, max: 0xFFFF })),
```

Comment:
Proposal: All uint8 and uint16 types we do that way? uint32 stay as UnsignedIntT and 64bit is already an other type?
I know that this is no topic in encoding, but so we enforce the types as in specs ...

What do you think?

Reply:
I think we need to define UInt8T, UInt16T, UInt32T and UInt64T and remove UnsignedIntT, BoundedUnsignedIntT, UnsignedLongT and BoundedUnsignedLongT.

min: 0 is already done since those values are unsigned
number maximum is higher the UInt32 so need to be bounded, bigint maximum is higher than UInt64 so also needs to be bounded.

I will send a PR in a 30mn to add this.

Reply:
I saw. I prepare  my side so that I can refresh PR when new types are existing ...

Reply:
(or as proposed .. we leave, merge and I send cleanup PRs, then we remove the old types). Up to you